### ISSUE TITLE
`ImplicitStringConcatenation` behaves differently with `ParserEngine: parser_prism`

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#12834]
## Expected behavior

I'm not sure 100% sure what the correct behaviour is.

## Actual behavior

Given this code:

```
<<~MESSAGE
  \n
  a
MESSAGE
```

The regular parser allows it, but with `parser_prism` it fails with:

`Combine  and "a\n" into a single string literal, rather than using implicit string concatenation.`

## RuboCop version

```                                                                                                                
1.63.1 (using Prism 0.24.0, rubocop-ast 1.31.2, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-graphql 1.5.1
  - rubocop-minitest 0.35.0
  - rubocop-performance 1.21.0
  - rubocop-rails 2.24.1
  - rubocop-sorbet 0.8.1
```

**Comment by andyw8** [comment]
Possibly related: https://github.com/ruby/prism/pull/2668

**Comment by koic** [comment]
This issue has been resolved in the latest Prism 0.30.
